### ISSUE TITLE
[type-annotation-spacing] return type in function shorthands treats like "property"

### ISSUE THREAD

**Issue by Ky6uk** [typescript-eslint/typescript-eslint#3640]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/type-annotation-spacing": ["error", {
      "before": true,

      "overrides": {
        "property": {
          "before": false
        }
      }
    }]
  }
}
```

```TS
const Test = defineComponent({
  name: 'Test',

  mounted() : void {
    //      ^--- Unexpected space before the ':'.
    console.log('Woof!');
  }
});
```

**Expected Result**

No errors. Return type in function shorthands should be treatened as `returnType`, not as `property`.

**Actual Result**

`4:13  error  Unexpected space before the ':'  @typescript-eslint/type-annotation-spacing`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.28.3` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.30.0` |
| `node`                             | `16.2.0` |

**Comment by Ky6uk** [comment]
This node has type `FunctionExpression` here, but the test for `property` uses `isClassOrTypeElement` that looks it includes lot of function-related node types.

**Comment by bradzacher** [comment]
This is working as intended - the same way it has worked since the option was first implemented back in #1496 over a year and a half ago.

Changing it would be a large breaking change, and given that nobody has complained, I think the majority of people are happy with it working this way. So I'm going to opt for leaving this as is.

**Comment by Ky6uk** [comment]
Or nobody is using this. I'm also using `typescript-eslint` for a couple of years but just decided to enable this rule because it was `off` and commented like a `bug` into our project. Well, the "bug" still exists and it looks the function return type will be treated as a "property" instead of "return type" for another couple of years. ¯\_(ツ)_/¯
